titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java

```diff
@@ -258,59 +313,106 @@ public void setVMLeaseHandler(Action1<List<? extends VirtualMachineLease>> lease
         this.leaseHandler = leaseHandler;
     }
 
+    @Override
+    public void setRescindLeaseHandler(Action1<List<LeaseRescindedEvent>> rescindLeaseHandler) {
+        this.rescindLeaseHandler = rescindLeaseHandler;
+    }
+
     @Override
     public Observable<LeaseRescindedEvent> getLeaseRescindedObservable() {
-        return vmLeaseRescindedObserver.asObservable();
+        return PublishSubject.create();
     }
 
     @Override
     public Observable<ContainerEvent> getTaskStatusObservable() {
         return vmTaskStatusObserver.asObservable();
     }
 
-    private void pollForNodes() {
+    private void getNodeUpdates() {
+        try {
+            long startTimeMs = clock.wallTime();
+            V1NodeList list = null;
+            try {
+                list = normalApi.listNode(null, null, null, null, null, null, null, null, null);
+                nodesClientMetrics.incrementOnSuccess(GET, NODES, STATUS_200);
+            } catch (Exception e) {
+                logger.error("Failed to list nodes with error:", e);
+                nodesClientMetrics.incrementOnError(GET, NODES, e);
+            } finally {
+                nodesClientMetrics.registerLatency(GET, startTimeMs);
+            }
+            if (list != null) {
+                Snapshot<V1Node> snapshot = Snapshot.newSnapshot(list.getItems(), clock.wallTime());
+                SnapshotComparison<V1Node> comparison = snapshot.compare(lastSnapshot, n -> n.getMetadata().getName(), (first, second) -> true);
```

Comment:
did you consider using `V1NodeList#metadata#resourceVersion` instead of having to compute a timestamp and detect changes manually?

Reply:
Yes, I plan on implementing the informer pattern similar to how k8s controllers work and it does this.